pkgs/build-support/rust/sysroot/default.nix

```diff
@@ -0,0 +1,47 @@
+{ stdenv, rust, rustPlatform, buildPackages }:
+
+{ shortTarget, originalCargoToml, target, RUSTFLAGS }:
+
+let rustSrc = stdenv.mkDerivation {
```

Comment:
Did you try using `rustPlatform.rustcSrc` for this?

Reply:
Yes. The problem is that `rustPlatform.rustcSrc` deletes the `tools` directory (see below). Should I add a parameter (defaulting to current behavior) to `rustcSrc` to determine if the tools directory is deleted)

https://github.com/NixOS/nixpkgs/blob/359ac19b17085a6957b48d8e409ebc0e8f3b1daf/pkgs/development/compilers/rust/rust-src.nix#L9